doc/autotest/Auto-test.md

```diff
@@ -0,0 +1,124 @@
+
+A property is tested in three stages: `make`, `run` and `post`. `make` prepare all computational tasks that are needed to calculate the property. For example to calculate EOS, `autotest` prepare a series of tasks, each of which has a scaled configuration with certain volume, and all necessary input files necessary for starting a VAPS or LAMMPS relaxation. `run` sends all the computational tasks to remote computational resources defined in a machine configuration file like `machine.json`, and automatically collect the results when remote calculations finish. `post` calculates the desired property from the collected results.
+
+### Relaxation
+
+The relaxation of a structure should be carried out before calculating all other properties:
+```bash
+dpgen autotest make equi.json 
+dpgen autotest run relax.json machine.json
+dpgen autotest post equi.json 
+```
+If, for some reason, the main program terminated at stage `run`, one can easily restart with the same command.
+`relax.json` is the parameter file. An example for `deepmd` relaxation is given as:
+```json
+{
+	"structures":	"confs/mp-*",
+	"interaction": {
+		"type":		"deepmd",
+		"model":	"frozen_model.pb",
+                "type_map":     {"Al": 0, "Mg": 1}
+	},
+	"relaxation": {
+	}
+}
+```
+
+where the key `structures` provides the structures to relax. `interaction` is provided with `deepmd`, and other options are `vasp`, `eam`, `meam`...
+
+Yuzhi:
+
+1. We should notice that the `interaction` here should always be considered as a unified abstract class, which means that we should avoid repeating identifing which interaction we're using in the main code.
+2. The structures here should always considered as a list, and the wildcard should be supported by using `glob`. Before all calculations , there is a stage where we generate the configurations.
+
+The outputs of the relaxation are stored in the `mp-*/00.relaxation` directory.
+```bash
+ls mp-*
+mp-1/relaxation  mp-2/relaxation  mp-3/relaxation
+```
+
+### Other properties
+
+Other properties can be computed in parallel:
+```bash
+dpgen autotest make properties.json 
+dpgen autotest run properties.json machine.json
+dpgen autotest post properties.json 
+```
+where an example of `properties.json` is given by
+```json
+{
+	"structures":	"confs/mp-*",
+	"interaction": {
+		"type":		"vasp",
+		"incar":	"vasp_input/INCAR",
+		"potcar_prefix":"vasp_input",
+		"potcars":	{"Al": "POTCAR.al", "Mg": "POTCAR.mg"}
+	},
+	"properties": [
+		{
+                        "type":         "eos",
+			"vol_start":	10,
+			"vol_end":	30,
+			"vol_step":	0.5
+		},
+		{
+                        "type":         "elastic",
+			"norm_deform":	2e-2,
+			"shear_deform": 5e-2
+		}
+        ]
+}
+```
```

Comment:
May compare this example with the current example in README. The example for eos here is not for the current autotest.